### PR TITLE
feat(invite-members): update sent invites ui

### PR DESCRIPTION
**Pull request by ameliahsu**
improve ui for after member invitations have been sent

before:
notice how all inputs are disabled after the user has pressed send. the user must click "send more invites" in order to reenable input and send additional invites. when they do click this button, all of their inputs are reset 🥲

https://github.com/user-attachments/assets/95030fae-ae4d-4310-af63-340b95d89f6c

after:
input is not disabled after invites have been sent 🥳 successfully sent invites are removed, while failed inputs are kept so that the user can view the error messages for each email. the user can continue to input additional emails without having to click an additional reset button. the status message in the bottom left is also _slightly_ different

https://github.com/user-attachments/assets/f54adf87-8eb0-430a-b0cf-4feff05b7033

### CODE REVIEW

File: static/app/components/modals/inviteMembersModal/inviteMembersFooter.tsx [hunk 1]
```diff
@@ -6,16 +6,34 @@ import InviteButton from 'sentry/components/modals/inviteMembersModal/inviteButt
...
   const isValidInvites = invites.length > 0;
 
+  const removeSentInvites = () => {
+    const statuses = Object.keys(inviteStatus);
+    statuses.forEach(status => {
+      if (!pendingInvites.emails.has(status)) {
+        delete inviteStatus[status];
+      }
+    });
+  };
```
Comment (JonasBa) [inline static/app/components/modals/inviteMembersModal/inviteMembersFooter.tsx:35]: We shouldn't be directly mutating context state as it wont dispatch react updates and could lead to weird behavior down the line as we are relying on some external rerender to update this.

**Review by JonasBa — COMMENTED** [review]
The old experience used to have that `add another` button which allowed for granular setting of access permissions per member, are we ok with removing that? It will make it more tedious to invite multiple people with different permissions, but maybe that is fine. 

Besides that, are users able to just input input multiple emails in order to invite multiple members now? If they can, I would add a small note hinting towards that

**Comment by ameliahsu** [comment]
@JonasBa Product decided that we wanted to remove the "Add another" option 👍 

The ability to input multiple emails already exists in the current design (but I am fixing copy/paste functionality in another PR). The placeholder text for email input says "Enter one more more emails" as a hint

**Review by JonasBa — APPROVED** [review]
Nice work with the context 👏🏼

**Pull request merged**